relentless/simulate/lammps.py

```diff
@@ -0,0 +1,290 @@
+        else:
+            cmds += ['region box block {} {} {} {} {} {}'.format(*box[:-3])]
+        cmds += ['create_box {N} box'.format(N=len(sim.ensemble.types))]
+
+        # use lammps random initialization routines
+        for i in sim.ensemble.types:
+            cmds += ['create_atoms {typeid} random {N} {seed} box'.format(typeid=sim.type_map[i],
+                                                                          N=sim.ensemble.N[i],
+                                                                          seed=self.seed+sim.type_map[i]-1)]
+        cmds += ['mass * 1.0',
+                 'velocity all create {temp} {seed}'.format(temp=sim.ensemble.T,
+                                                            seed=self.seed)]
+
+        cmds += self.attach_potentials(sim)
+
+        return cmds
+
+class MinimizeEnergy(simulate.SimulationOperation):
+    def __init__(self, energy_tolerance, force_tolerance, max_iterations):
+        self.energy_tolerance = energy_tolerance
+        self.force_tolerance = force_tolerance
+        self.max_iterations = max_iterations
+
+    def to_commands(self, sim): #TODO: min_style, min_modfy?
+        cmds += ['minimize {etol} {ftol} {maxiter}'.format(etol=self.energy_tolerance,
+                                                           ftol=self.force_tolerance,
+                                                           maxiter=self.max_iterations)]
+
+        return cmds
+
+# Brownian dynamics not supported by LAMMPS
+
+class AddLangevinIntegrator(simulate.SimulationOperation):
+    def __init__(self, idx, group_idx, t_start, t_stop, damp, seed): #TODO: condense keywords, reconcile with generic?
+        self.idx = idx
+        self.group_idx = group_idx
+        self.t_start = t_start
+        self.t_stop = t_stop
+        self.damp = damp
+        self.seed = seed
+
+    def to_commands(self, sim):
+        cmds += ['fix {idx} {group_idx} langevin {t_start} {t_stop} {damp} {seed}'.format(idx=self.idx,
+                                                                                          group_idx=self.group_idx,
+                                                                                          t_start=self.t_start,
+                                                                                          t_stop=self.t_stop,
+                                                                                          damp=self.damp,
+                                                                                          seed=self.seed)]
+
+        return cmds
+
+class RemoveLangevinIntegrator(simulate.SimulationOperation):
+    def __init__(self, add_op):
+        if not isinstance(add_op, AddLangevinIntegrator):
+            raise TypeError('Addition operation is not AddLangevinIntegrator.')
+        self.add_op = add_op
+
+    def to_commands(self, sim):
+        cmds += ['unfix {idx}'.format(idx=self.add_op.idx)]
```

Comment:
Unfix based on the fix ID you automatically assign. Also, you will need to unfix both the NVE integrator and the langevin fixes.